File: Homeworks/C#1/03.OperatorsAndExpressionsHomework/11.ExtractThirdBit/ExtractThirdBit.cs
using System;

class ExtractThirdBit
{
    static void Main(string[] args)
    {
        Console.Write("Please, enter unsigned integer: ");
        int number = int.Parse(Console.ReadLine());
        Console.Clear();
        Console.WriteLine("The binary representation of the {0} is:\n{1}"
                            , number, Convert.ToString(number, 2).PadLeft(16, '0'));
        int p = 3;
        int mask = 1 << p;
        int numberAndMask = number & mask;
        int bitResult = numberAndMask >> p;

            Console.WriteLine("The third bit is:" + bitResult);
    }
}

File: Homeworks/JavaScriptFundamentals/02.OperatorsAndExpressions/05.ExtractThirdBit/ExtractThirdBit.js
/*
 Write a boolean expression for finding if the bit 3
 (counting from 0) of a given integer is 1 or 0.
 */

var jsConsole;
var button = document.getElementById('check');

button.onclick = extractThirdBit;

function extractThirdBit(){
    var number = jsConsole.readInteger('input');
    var binary = number.toString(2);

    var mask = 1 << 3;
    var maskAndNumber = number & mask;
    var result = maskAndNumber >> 3;

    jsConsole.writeLine('Binary representation of input number is: ' + binary);
    jsConsole.writeLine('Third bit is: ' + result);

    console.log('Binary representation of input number is: ' + binary);
    console.log('Third bit is: ' + result);
}